pvacseq/lib/vaccine_design.py

```diff
@@ -0,0 +1,378 @@
+                    peptides[key]
+                except KeyError:
+                    keyErrorCount += 1
+                    continue
+
+                if seqID not in iedb_results:
+                    iedb_results[seqID] = {}
+                allele = peptides[key]['allele']
+                if allele not in iedb_results[seqID]:
+                    iedb_results[seqID][allele] = {}
+                    if 'total_score' not in iedb_results[seqID][allele]:
+                        iedb_results[seqID][allele]['total_score'] = list()
+                        iedb_results[seqID][allele]['total_score'].append(peptides[key]['ic50'])
+                    else:
+                        iedb_results[seqID][allele]['total_score'].append(peptides[key]['ic50'])
+
+                if 'min_score' in iedb_results[seqID][allele]:
+                    iedb_results[seqID][allele]['min_score'] = min(iedb_results[seqID][allele]['min_score'], peptides[key]['ic50'])
+                else:
+                    iedb_results[seqID][allele]['min_score'] = peptides[key]['ic50']
+                    successCount += 1
+
+    print("Successful ic50 mappings: " + str(successCount) + " errors: " + str(keyErrorCount))
+
+    Paths = nx.DiGraph()
+    spacers = [None, "HH", "HHC", "HHH", "HHHD", "HHHC", "AAY", "HHHH", "HHAA", "HHL", "AAL"]
+    for ep in combinations:
+        ID_1 = ep[0]
+        ID_2 = ep[1]
+        Paths.add_node(ID_1)
+        Paths.add_node(ID_2)
+        for space in spacers:
+            if space is None:
+                key = str(ID_1 + "|" + ID_2)
+            else:
+                key = str(ID_1 + "|" + space + "|" + ID_2)
+            worst_case = sys.maxsize
+            for allele in iedb_results[key]:
+                if iedb_results[key][allele]['min_score'] < worst_case:
+                    worst_case = iedb_results[key][allele]['min_score']
+            if Paths.has_edge(ID_1, ID_2) and Paths[ID_1][ID_2]['weight'] < worst_case:
+                Paths[ID_1][ID_2]['weight'] = worst_case
+                if space is not None:
+                    Paths[ID_1][ID_2]['spacer'] = space
+                else:
+                    Paths[ID_1][ID_2]['spacer'] = ''
+            elif not Paths.has_edge(ID_1, ID_2):
+                if space is not None:
+                    Paths.add_edge(ID_1, ID_2, weight=worst_case, spacer=space)
+                else:
+                    Paths.add_edge(ID_1, ID_2, weight=worst_case, spacer='')
+
+    print("Graph contains " + str(len(Paths)) + " nodes and " + str(Paths.size()) + " edges.")
+    print("Finding path.")
+
+    distance_matrix = {}
+    for ID_1 in Paths:
+        try:
+            distance_matrix[ID_1]
+        except KeyError:
+            distance_matrix[ID_1] = {}
+        for ID_2 in Paths[ID_1]:
+            distance_matrix[ID_1][ID_2] = Paths[ID_1][ID_2]['weight']
+
+    init_state = seq_keys
+    #random.shuffle(init_state)      # This line helps, but makes unit testing very difficult
```

Comment:
The usual way to get around this is to have the test provide a seed to the RNG so that the output is identical for test runs